### ISSUE TITLE
How to confirm Order in CyberSource.

### ISSUE THREAD

**Issue by poonam-Bacancy** [spree/spree#9485]
I have a couple of issue when i am trying to setup CyberSource payment gateway in one of my spree project. :- 

(1) It is directly goes to the "Complete( Receipt Page )" Page after " Payment " page  instead of  "Confirmation" page.
(2) It is not allowing to use existing card in CyberSource payment gateway.

In checkout process , 
(1) You have to first add delivery address Then 
(2) You have to add the shipping method Then 
(3) You are going to the " Payment " step and you are selecting the "New Credit Card " option and fill all the required details AND 
when you click on NEXT it should directly "Complete" the order without "Confirming" the order. 

We need to first " Confirm " the order then it should goes to the "Complete " step.

version used: 
spree Version :- 3.4.4
Spree Gateway :- 3.3

**Comment by nvh0412** [comment]
@poonam-Bacancy The root cause is `CyberSource` payment method is being defined in Spree-gateway gem, and based on its implementation, it doesn't support `payment profiles` mechanism and Spree is set `false` value to `always_include_confirm_step` in setting by default, so all these stuff leads you guys to ignore the confirmation view.

First, please make sure that you guys know the prerequisite of showing the confirmation view, if not, you guys can take a look at this

https://github.com/spree/spree/blob/c6dd7d3c0817e3be5f657a62bd3def8ff4ec39c2/core/app/models/spree/order.rb#L202-L203

So basically, if you want your spree app to alway show the confirmation view, then please consider adding this stuff to your spree initializer file

```ruby
Spree.config do |config|
  config.always_include_confirm_step = true
end
```

**Comment by poonam-Bacancy** [comment]
@nvh0412  Yes, with the help of config.always_include_confirm_step = true it displays the "Confirmation" view page But when we click on the "Place order" button it will redirect back to the "Payment" step instead of the "Complete" step. 

So If I add forcefully the confirmation step like this then it is giving me following error :

Gateway Error
  --- !ruby/object:ActiveMerchant::Billing::Response
params:
  merchantReferenceCode: R838448420-PKEZ3BB9
  requestID: '5671594226706420304008'
  decision: REJECT
  reasonCode: '101'
  message: Request is missing one or more required fields
  missingField: c:card/c:accountNumber
  requestToken: AhjzrwSTM3UuY8ugd4iIEQFRp/JIAM+kC4YSDSMukmXoxXxvhJMzdS5jy6B3iIgAkhXF
message: Request is missing one or more required fields
success: false
test: true
authorization: 
fraud_review: 
error_code: 
emv_authorization: 
avs_result:
  code: 
  message: 
  street_match: 
  postal_match: 
cvv_result:
  code: 
  message: 

Can you please guide me how to handle this error ?
